tests/conftest.py

```diff
@@ -16,6 +16,13 @@
 
 sys.path.append(str(TEST_ROOT.parent))
 
+build_artifacts = Path("/dev/null")  # folder with cargo-built executables
+
+
+def cargo_build() -> Path:
+    subprocess.run(["cargo", "build"])
+    return Path("target/debug")
```

Comment:
Does this create a new build directory in ./tests?

```suggestion
    return TEST_ROOT.parent.joinpath("target", "debug")
```

`cwd=TEST_ROOT.parent` in the build command might help as well.

Reply:
`cargo build` builds into vmsh/target/debug when called from somewhere within `vmsh/`.

Reply:
You are probably right. Anyway does not hurt to hard-code it.